pkg/handler/collector/collector_test.go

```diff
@@ -147,10 +159,22 @@ func testSubscribe(ctx context.Context, transportFunc func(processor.DocumentTre
 	}
 
 	processFunc := func(d []byte) error {
+
+		blobStoreKey, err := events.DecodeEventSubject(ctx, d)
+		if err != nil {
+			logger.Errorf("[processor: %s] failed decode event: %v", uuidString, err)
+			return nil
+		}
+
+		documentBytes, err := blobStore.Read(ctx, blobStoreKey)
+		if err != nil {
+			return fmt.Errorf("failed read document to blob store: %w", err)
+		}
```

Comment:
i think it would be helpful to abstract this where this should just deal with the document/blob struct instead of being exposed to the blob store internals?

Reply:
this is already abstracted from the blob store internals. It calls the `blobStore` struct `Read` method